openedx/core/djangoapps/programs/utils.py

```diff
@@ -251,25 +251,13 @@ def _is_course_complete(self, course):
         Returns:
             bool, indicating whether the course is complete.
         """
-
-        def reshape(course_run):
-            """
-            Modify the structure of a course run dict to facilitate comparison
-            with course run certificates.
-            """
-            return {
-                'course_run_id': course_run['key'],
-                # A course run's type is assumed to indicate which mode must be
-                # completed in order for the run to count towards program completion.
-                # This supports the same flexible program construction allowed by the
-                # old programs service (e.g., completion of an old honor-only run may
-                # count towards completion of a course in a program). This may change
-                # in the future to make use of the more rigid set of "applicable seat
-                # types" associated with each program type in the catalog.
-                'type': course_run['type'],
-            }
-
-        return any(reshape(course_run) in self.completed_course_runs for course_run in course['course_runs'])
+        for course_run in course['course_runs']:
```

Comment:
I think it's better to keep the original `reshape` function.
All you need to do is, extra the `course_run['type']` out and assign it into a new variable.
like this:
```
course_type = course_run['type']
if course_type=='credit':
    ### Add comments on why we want to do this
    course_type='verified'

return {
    'course_run_id': course_run['key'],
    'type': course_type,
}
```